src/freenet/clients/fcp/ClientPut.java

```diff
@@ -527,17 +530,19 @@ RequestStatus getStatus() {
 				total = (int) msg.getTotalBlocks();
 				min = (int) msg.getMinBlocks();
 				fetched = (int) msg.getFetchedBlocks();
+				latestSuccess = msg.getLatestSuccess();
 				fatal = (int) msg.getFatalyFailedBlocks();
 				failed = (int) msg.getFailedBlocks();
+				latestFailure = msg.getLatestFailure();
 				totalFinalized = msg.isTotalFinalized();
 			}
 		}
 		
-		return new UploadFileRequestStatus(identifier, persistence, started, finished, 
-				succeeded, total, min, fetched, fatal, failed, totalFinalized, 
-				lastActivity, priorityClass, finalURI, uri, failureCode,
-				failureReasonShort, failureReasonLong, getDataSize(), mimeType,
-				fnam, isCompressing());
```

Comment:
May I register my displeasure that these arguments are all broken apart? This is a huge list, and this change perpetuates it. (Not that its intent was to clean it up.)